sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1050,6 +1050,13 @@ private boolean isManagementToken(String lockToken) {
             });
     }
 
+    Mono<ServiceBusReceiverAsyncClient> createConsumerWithReceiveLink() {
+        return Mono.defer(() -> {
+            getOrCreateConsumer();
```

Comment:
One last question. So we get and forget the returned result?

Reply:
This would be saved in existing `consumer` in this instance and will be used for all next calls.
https://github.com/Azure/azure-sdk-for-java/pull/17227/files#diff-42c5fc794930edf5858987b570b871727693454acb27130ac5a9748e55d2de95L1054

Reply:
This will be saved on in class instance `consumer` and returned next time.